plugins/foreman_remote_execution/0.1/index.md

```diff
@@ -0,0 +1,394 @@
+
+    foreman-installer --enable-foreman-plugin-remote-execution\
+                      --enable-foreman-proxy-plugin-remote-execution-ssh
+
+The installer will automatically configure an SSH key for you, or you may configure an existing see.  See the foreman-installer --help for more information.
+
+### Manual Installation
+
+In the case you want to install the package manually on an existing Foreman or
+a Katello, use the following steps.
+
+Install the packages appropriate for your operating system:
+
+RPM:
+
+    yum install tfm-rubygem-foreman_remote_execution tfm-rubygem-foreman-tasks\
+                rubygem-smart_proxy_remote_execution_ssh
+
+DEB:
+
+    apt-get install ruby-foreman-tasks ruby-foreman-remote-execution\
+                    ruby-smart-proxy-remote-execution-ssh
+
+Next you have to setup ssh keys. By default smart proxy loads the key
+from `/usr/share/foreman-proxy/.ssh/id_rsa_foreman_proxy`. To customize it you
+can edit the configuration in
+`/etc/foreman-proxy/config/settings.d/remote_execution_ssh.yml`. Without
+customization you need to create new ssh key and distribute it to
+target hosts. The key must not use passphrase.
+
+To generate a key, run following command on the host where Smart proxy runs
+
+    mkdir ~foreman-proxy/.ssh
+    chown foreman-proxy ~foreman-proxy/.ssh
+    sudo -u foreman-proxy ssh-keygen -f ~foreman-proxy/.ssh/id_rsa_foreman_proxy -N ''
+
+When using SELinux make sure the directory and the files have correct labels
+of ssh_home_t. If not, restore the context:
+
+    restorecon -RvF ~foreman-proxy/.ssh
+
+Don't forget to restart Foreman, Smart proxy and Foreman tasks so
+plugins are loaded
+
+    service httpd restart
+    service foreman-tasks restart
+    service foreman-proxy restart
+
+Finally, you have to refresh the Smart proxy features in the Foreman:
+Dynflow and Ssh should appear there as new features of the Smart proxy.
+
+## 2.2 Remote Hosts Configuration
+
+The remote hosts need to be configured to accept the private key that the smart
+proxy is using. For now, `root` is used as the user to access the remote hosts.
+It will be possible to change this in later versions.
+
+The ssh keys for the smart proxies are available as a host parameter -
+`remote_execution_ssh_keys`.  This allows you to manage the authorized keys
+with your configuration management platform of choice, or through a
+provisioning template.
+
+If you prefer to use a provisioning template, an example snippet is included.
+In the kickstart %post or finish template, simply include:
+
+     <%= snippet 'remote_execution_ssh_keys' %>
+
+If you prefer to distribute the keys manually, you may do this:
+
+    ssh-copy-id -i ~foreman-proxy/.ssh/id_rsa_foreman_proxy.pub root@target.example.com
+
+Alternatively, the Foreman proxy publishes the public key it over the
+API, so that one can just download it into authorized keys:
+
+    curl https://myproxy.example.com:8443/ssh/pubkey >> ~/.ssh/authorized_keys
```

Comment:
Is this really Foreman or Smart-proxy API? Looks like Smart-proxy. I'd rather use "Foreman Proxy" in the text above, each of these can have a different key.